gateway/src/state.rs

```diff
@@ -481,6 +490,73 @@ impl State {
     pub fn noop_msg(&self) {
         panic!("error committing cross-msg. noop should be returned but not implemented yet");
     }
+
+    /// Insert a cross message to the `postbox` before propagate can be called for the
+    /// message to be propagated upwards or downwards.
+    ///
+    /// # Arguments
+    /// * `st` - The blockstore
+    /// * `owners` - The owners of the message that started the cross message. If None means
+    ///              anyone can propagate this message. Allows multiple owners.
+    /// * `gas` - The gas needed to propagate this message
+    /// * `msg` - The actual cross msg to store in `postbox`
+    pub(crate) fn insert_postbox<BS: Blockstore>(
+        &mut self,
+        st: &BS,
+        owners: Option<Vec<Address>>,
+        gas: TokenAmount,
```

Comment:
We can remove `gas` from the postbox item, as discussed this is set by the actor.